exodus_lambda/functions/base.py

```diff
@@ -1,18 +1,35 @@
 import json
+import logging.config
+import logging
 
 
 class LambdaBase(object):
-    def __init__(self, conf_file="lambda_config.json"):
+    def __init__(self, logger_name="default", conf_file="lambda_config.json"):
         self._conf_file = conf_file
         self._conf = None
+        self._logger_name = logger_name
+        self._logger = None
 
     @property
     def conf(self):
         if not self._conf:
             with open(self._conf_file, "r") as json_file:
                 self._conf = json.load(json_file)
-
         return self._conf
 
+    @property
+    def logger(self):
+        if not self._logger:
+            log_conf = self.conf.get("logging", {})
+            logging.config.dictConfig(log_conf)
+            root_logger = logging.getLogger()
+            self._logger = logging.getLogger(self._logger_name)
+            if log_conf:
```

Comment:
```suggestion
            if log_conf and root_logger.handlers:
```

In a clean Python environment, the root logger starts with no handlers, so I think this should probably have a check for root_logger.handlers and not proceed if it's empty.

```
$ python3
Python 3.7.5 (default, Dec 15 2019, 17:54:26) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> logging.getLogger().handlers
[]
```

The Lambda runtime seems to always install a handler, but I think it's better for the code not to crash if that didn't happen, mainly for the sake of testing.  If tests right now are not crashing here, I guess that's only because something in the test framework has added a handler on the root logger?  Seeing as I don't know where that's coming from, I'd prefer us not to rely on it.

Reply:
Good call. Apparently, if one does not add a `-p no:logging` arg to pytest, pytest adds a `<LogCaptureHandler (NOTSET)>` handler to the root logger for/by the caplog fixture.

Reply:
> I think this should probably have a check for root_logger.handlers and not proceed if it's empty

@rohanpm , I wasn't sure if by 'not proceed' if you meant to gracefully except an error, or to add a handler.

Reply:
I just mean don't enter this block, i.e. if there is a handler then configure it, if there is no handler then do nothing.

The primary concern is just to not crash so that we've not added a precondition onto this code, "root logger must have at least one handler to use this class".